polyply/src/apply_modifications.py

```diff
@@ -12,5 +12,96 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-class ApplyModifications():
-      pass
+from vermouth.log_helpers import StyleAdapter, get_logger
+from vermouth.processors.annotate_mut_mod import parse_residue_spec
+LOGGER = StyleAdapter(get_logger(__name__))
+from .processor import Processor
+import networkx as nx
+
+def _get_protein_termini(meta_molecule):
+    """
+    make a resspec for a protein with correct N-ter and C-ter
+    """
+    max_resid = meta_molecule.max_resid
+    last_node = max_resid - 1
+    last_resname = meta_molecule.nodes[last_node]['resname']
+
+    protein_termini = [({'resid': 1, 'resname': meta_molecule.nodes[0]['resname']}, 'N-ter'),
+                       ({'resid': max_resid, 'resname': last_resname}, 'C-ter')
+                       ]
+
+    return protein_termini
```

Comment:
What do you think about replacing this with something like:

```python
def _patch_protein_termini(meta_molecule, ter_mods=['C-ter', 'N-ter']):
    """
    make a resspec for a protein with correct terminal modification
    """
    protein_termini = [({'resid': 1, 'resname': meta_molecule.nodes[0]['resname']}, ter_mods[0])]
    if len(ter_mods) > 1:
        max_resid = meta_molecule.max_resid
        last_node = max_resid - 1
        last_resname = meta_molecule.nodes[last_node]['resname']
        last_mod = ({'resid': max_resid, 'resname': last_resname}, ter_mods[1])
        protein_termini.append(last_mod)

    return protein_termini
```

ter_mods can then be: `['N-ter', 'C-ter']` or `['NH2-ter', 'COOH-ter']` or `['Zwitter']` 

of course, the argparser needs to be changed accordingly to accept one or two values. The default would be  `['N-ter', 'C-ter']`. This way we always get the correct default termini.